server/aspnetcore/TicketHub/Program.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.AspNetCore.Diagnostics;
+using Newtonsoft.Json;
+using TicketHub.Database;
+using Styra.Opa;
+using Styra.Opa.AspNetCore;
+using Microsoft.AspNetCore.Authentication;
+
+
+string opaURL = System.Environment.GetEnvironmentVariable("OPA_URL") ?? "http://localhost:8181";
+OpaClient opa = new OpaClient(opaURL);
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the builder.
+builder.Services.AddDbContext<PostgresContext>();
+builder.Services.AddLogging(builder => builder.SetMinimumLevel(LogLevel.Trace).AddConsole());
+builder.Services.AddControllers().AddNewtonsoftJson(opts =>
+{
+    opts.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+    opts.SerializerSettings.NullValueHandling = NullValueHandling.Ignore;
+});
+builder.Services.AddExceptionHandler<CustomExceptionHandler>();
+builder.Services.AddAuthentication(options =>

```

Comment:
[server change]: This is where the custom AuthN stuff had to be plugged in. Without this, the `OpaAuthorizationMiddleware` will default-deny *all* incoming requests as unauthenticated.